qa/tasks/rados.py

```diff
@@ -272,9 +279,16 @@ def thread():
                     )
                 tests[id_] = proc
             run.wait(tests.values())
+            wait_for_all_active_clean_pgs = config.get("wait_for_all_active_clean_pgs", False)
+            # usually set when we do min_size testing.
+            if  wait_for_all_active_clean_pgs:
+                # Make sure we finish the test first before deleting the pool.
+                # Mainly used for test_pool_min_size
+                manager.wait_for_clean()
+                manager.wait_for_all_osds_up(timeout=300)
```

Comment:
I can't think of any rados task users that wouldn't want this check.  Do you have any examples?

Reply:
So before I made this PR (change test_pool_min_size to only run for ~30mins by default), I know that `test_pool_min_size` actually will forever until the `rados task` finishes first & delete all the pools, so if I didn't make test_pool_min_size finite, then the test would have been stuck forever since test_pool_min_size will keep on killing and reviving OSDs.  Therefore, I didn't want to enforce this flag on all the tests in fear of affecting other tests that uses rados task.

Reply:
I see, generally the background thrasher continues until the tasks above it stop so this wait_for_all_osds_up won't in generally work.  I don't actually understand what makes test_pool_min_size different in this regard -- why is it important that it complete before the rados task?

Reply:
@athanatos 
So it doesn't necessary need to complete before the rados task finishes it's workload, but rados task needs to wait for `test_pool_min_size` to finish before rados task can start deleting pools because there might be a scenario in `test_pool_min_size` where we are still waiting for PGs to become active+clean, but then rados task deletes the pool and suddenly the test passed without actually knowing if the PGs will actually recover or not.